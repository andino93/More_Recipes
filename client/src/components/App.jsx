import React from 'react';
import PropTypes from 'react-proptypes';
import HeaderComponent from './partials/Headers/Header.jsx';
/**
 * App class declaration
 */
export default class App extends React.Component {
  /**
   * Renders the component
   * @return {XML} XML/JSX
   */
  render() {
    return (
      <div>
        {/*<HeaderComponent/>*/}
        { this.props.children }
      </div>
    );
  }
}
App.propTypes = {
  children: PropTypes.array.isRequired
};
