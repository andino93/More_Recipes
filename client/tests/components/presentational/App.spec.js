import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import localStorageMock from '../../__mocks__/localStorageMock';
import App from '../../../src/components/App.jsx';
import mockData from "../../__mocks__/mockData";

configure({ adapter: new Adapter() });

// noinspection JSAnnotator
window.localStorage = localStorageMock;
window.localStorage.token = mockData.token;

describe('<App>', () => {
  const store = mockStore({});
  const connectedHomepage = mountWrap(App, store, {
    children: []
  });
  it('should render the UserRecipe Component', (done) => {
    expect(toJson(connectedHomepage)).toMatchSnapshot();
    done();
  });
});