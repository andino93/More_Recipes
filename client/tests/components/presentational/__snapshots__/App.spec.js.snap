// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should render the UserRecipe Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <App>
        <div />
      </App>
    </Router>
  </BrowserRouter>
</Provider>
`;
